client/src/main/java/io/pravega/client/netty/impl/RawClient.java

```diff
@@ -131,12 +140,26 @@ private void closeConnection(Throwable exceptionToInflightRequests) {
     }
 
     public <T extends Request & WireCommand> CompletableFuture<Reply> sendRequest(long requestId, T request) {
+        return sendRequest(requestId, request, this.connectionTimeout);
+    }
+
+    public <T extends Request & WireCommand> CompletableFuture<Reply> sendRequest(long requestId, T request,
+                                                                                  Duration timeout) {
         return connection.thenCompose(c -> {
             log.debug("Sending request: {}", request);
             CompletableFuture<Reply> reply = new CompletableFuture<>();
             synchronized (lock) {
                 requests.put(requestId, reply);
             }
+            CompletableFuture<Void> timer = Futures.futureWithTimeout(timeout, connectionFactory.getInternalExecutor());
```

Comment:
This pattern of creating a future with a timeout and scheduling a job after a timeout, would work better if combined into a single method similar to this one: https://github.com/pravega/pravega/pull/4698/files#diff-0bcfd93b4074110157f7e0729d9c1f11R250

It would be good to add that to Futures and update anywhere else using those two calls in succession to use the new method.

Reply:
Got it